rust/main/agents/relayer/src/test_utils/mock_base_builder.rs

```diff
@@ -21,6 +21,13 @@ pub struct MockBaseMetadataBuilderResponses {
     pub get_proof: ResponseList<eyre::Result<Proof>>,
     pub highest_known_leaf_index: ResponseList<Option<u32>>,
     pub get_merkle_leaf_id_by_message_id: ResponseList<eyre::Result<Option<u32>>>,
+    /// build_ism uses a hashmap of VecDeque responses instead.
+    /// This is because AggregationISMs run in parallel, so having just
+    /// a single VecDeque shared between the different threads yields unpredictable
+    /// results. One thread might run for a long time and starve the other threads
+    /// of responses.
+    /// In order to fix this in tests, the mock ISMs are built with specific addresses
+    /// in place. And gets responses from this based on the address.
```

Comment:
nice that's useful to know. Wondering when you say "uses a hashmap of VecDeque responses instead", instead of what it is you mean? Assuming a concurrent hashmap instead of an arcmutex. But since the mutex is still locked across the entire hashmap and the tests run fine I think it's not starvation that's the issue? More likely tests overwriting each other's state - we should update the comment to reflect this if that's the case

Reply:
ah, what I meant by instead is because all the other responses still just uses an `Arc<Mutex<VecDeque>>`.
If we need more complicated behaviour in the future, we can always come back and change them to a hashmap or some other data structure that suits our needs.

Reply:
I see, can you update the comment to clarify pls. This point also still stands I believe, so should the starvation point should be replaced:
> But since the mutex is still locked across the entire hashmap and the tests run fine I think it's not starvation that's the issue? More likely tests overwriting each other's state (vecdeque) - we should update the comment to reflect this if that's the case